bin/node/cli/tests/export_import_flow.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+
+use assert_cmd::cargo::cargo_bin;
+use std::{process::Command, fs};
+use tempfile::tempdir;
+
+mod common;
+
+#[test]
+fn export_default_import_default_works() {
+	let base_path = tempdir().expect("could not create a temp dir");
+	let exported_blocks = base_path.path().join("exported_blocks");
+
+	common::run_dev_node_for_a_while(base_path.path());
+
+	let status = Command::new(cargo_bin("substrate"))
+		.args(&["export-blocks", "--pruning", "archive", "--dev", "-d"])
+		.arg(&base_path.path())
+		.arg(&exported_blocks)
+		.status()
+		.unwrap();
+	assert!(status.success());
+
+	let metadata = fs::metadata(&exported_blocks).unwrap();
+	assert!(metadata.len() > 0, "file exported_blocks should not be empty");
+
+	let _ = fs::remove_dir_all(base_path.path().join("db"));
+
+	let status = Command::new(cargo_bin("substrate"))
+		.args(&["import-blocks", "--pruning", "archive", "--dev", "-d"])
+		.arg(&base_path.path())
+		.arg(&exported_blocks)
+		.status()
+		.unwrap();
+	assert!(status.success());
+}
```

Comment:
This test looks duplicated with:

https://github.com/paritytech/substrate/blob/6d2ce02cf94200ce952c3732938fb500e3593624/bin/node/cli/tests/import_export_and_revert_work.rs#L25-L59

Can you remove one or the other? (Don't forget to do the revert as it needs to tested also)

Reply:
So I indeed noticed that they looked very similar (I actually first copied the code from the other test). However, there are a couple of points I'd like to address:
1. Currently, running `export-blocks` and then `import-blocks` fail (as mentioned in the PR description). This means that we have no regression test. Indeed the test you linked only `assert!(status.success())` which doesn't check for error in the log output.
2. I think we should make sure that the `--binary` flag works too :)

In my mind, we would have `import_export_and_revert_work.rs` which would actually test the `revert` mechanism. We would keep the tests I've written (and add two more, namely `export_default_import_json_fails` and `export_json_import_default_fails`), but use [`command::output`](https://doc.rust-lang.org/std/process/struct.Command.html#method.output) to monitor the log errors, as to make sure that the regression test works correctly! 

What do you think? :)

Reply:
> they need to run the node for a certain amount of time every time

Can't we move that into a single place, store the generated database artefacts and just re-use that for these tests? That would speed up a lot.

Reply:
Fair point, I didn't think about that :thinking: 

This is out of scope of this PR but it gives an opposite argument.

Reply:
Good idea.. Not sure how to move from here though. Should I add the 4 tests I mentioned earlier, and create an associated issue? I'd be happy to jump on the associated issue once this PR gets merged in :

Reply:
I think it's best to merge as it for now and make a new PR with the 4 tests and the idea of @gnunicorn so we don't end up in a state where we have a lot more test but no speed optimization.

Reply:
Deal! Last question @cecton you metioned calling `revert` too, but that was not the original idea of the tests I wrote. Should I still add it?

Reply:
I personally think it would be best for a question of CI speed but I let you decide what you believe is best.

Reply:
> It is all good ideas but there is a huge problem with those CLI tests: they are extremely expensive in term of CI time. (They need to run sequentially + they are more prone to be flaky + they need to run the node for a certain amount of time every time) Overall I see them more like small "test scenarios".

So I just ran the four tests, and it looks like they're not being run sequentially. Since they're all using their own `tmp` directory, I think we're fine having four tests, right?

Reply:
So after a long day of work I successfully managed to over-engineer those tests....
Please have a look at https://github.com/paritytech/substrate/pull/5816/commits/c986b440143400933e148178dba8aa6661b56d53 and tell me if we should revert it or not :)